src/modules/flow/netctl/netctl.json

```diff
@@ -0,0 +1,139 @@
+{
+  "$schema": "http://solettaproject.github.io/soletta/schemas/node-type-genspec.schema",
+  "name": "netctl",
+  "meta": {
+    "author": "Intel Corporation",
+    "license": "Apache-2.0",
+    "version": "1"
+  },
+  "types": [
+    {
+      "category": "netctl",
+      "description": "Send list of all network services attached to board.",
+      "methods": {
+        "open": "open_list_services",
+        "close": "close_list_services"
+      },
+      "name": "netctl/list-services",
+      "out_ports": [
+        {
+          "data_type": "string",
+          "description": "Name of network services supported by system.",
+          "name": "OUT"
```

Comment:
`OUT` is not enough. You need two ports:
 - `ADDED` services that are new.
 - `REMOVED` services that were removed.

No need to keep an internal shadow list, you do not need a private data for this node type.